docs/api/synchronization_library/barrier.md

```diff
@@ -23,26 +23,35 @@ cuda::std::barrier<> bb;
 cuda::barrier<cuda::thread_scope_block> c;
 ```
 
-The class template `barrier` may also be declared without initialization; a
+The class template `barrier` may also be declared without initialization in the `cuda::` namespace; a
 friend function `init` may be used to initialize the object.
 
 ```c++
 // Shared memory does not allow initialization.
 __shared__ cuda::barrier<cuda::thread_scope_block> b;
 
 init(&b, 1); // Use this friend function to initialize the object.
+/*
+namespace cuda {
+  template<thread_scope Sco, class CompletionF>
+  __host__ __device__ void init(barrier<Sco,CompletionF>* bar, std::ptrdiff_t expected);
+  template<thread_scope Sco, class CompletionF>
+  __host__ __device__ void init(barrier<Sco,CompletionF>* bar, std::ptrdiff_t expected, CompletionF completion);
+}
+*/
 ```
 
+- Expects: `*bar` is trivially initialized.
```

Comment:
If the intent is to exhibit undefined behavior if this condition is not met: 

```suggestion
- Preconditions: `*bar` is trivially initialized.
```

See: https://eel.is/c++draft/structure.specifications#3